codegen/src/types.rs

```diff
@@ -1046,31 +1052,35 @@ pub struct FileDescriptor {
 
 impl FileDescriptor {
 
-    pub fn write_proto<P: AsRef<Path>>(in_file: P, out_file: P) -> Result<()> {
-        let mut desc = FileDescriptor::read_proto(&in_file)?;
-        desc.fetch_imports(in_file.as_ref())?;
+    pub fn write_proto<P: AsRef<Path>>(config: &Config<P>) -> Result<()> {
```

Comment:
Design-wise I'd prefer `Config` to be a simple struct with no additional functionality outside of mangling the configuration values (perhaps some of the path mangling in `write_proto` could move there). Assuming later on with more options, it will be borrowed deeper and deeper into the methods/functions doing the actual job. While I do not yet see much reason for making `FileDescriptor` private perhaps it could be managed with a free function like below instead?

```
pub fn write_proto<P: AsRef<Path>>(config: &Config<P>) -> Result<()> {
  // ...
}
```